src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -65,9 +111,30 @@ public PolicyEntity createPolicy(@PathParam("domainid") Long domainid, PolicyEnt
     public PolicyEntity updatePolicy(@PathParam("domainid") Long domainid, @PathParam("policyid") Long policyid,
             PolicyEntity entity) {
 
-        logger.info("Updating Policy Entity ...");
-        //TODO Sudhir: Add db calls here
-        return null;
+        logger.info("Updating for Policy Entity Id...:" + policyid);
+
+        return this.txControl.required(new Callable<PolicyEntity>() {
+
+            @Override
+            public PolicyEntity call() throws Exception {
+
+                CriteriaBuilder cb = em.getCriteriaBuilder();
+
+                CriteriaQuery<PolicyEntity> q = cb.createQuery(PolicyEntity.class);
+                Root<PolicyEntity> r = q.from(PolicyEntity.class);
+                q.select(r).where(cb.and(cb.equal(r.get("domain").get("Id"), domainid)),
+                        cb.equal(r.get("id"), policyid));
+
+                List<PolicyEntity> result = em.createQuery(q).getResultList();
+                if (result.isEmpty() == false) {
+                    result.get(0).setName(entity.getName());
+                    em.persist(result.get(0));
+                    return entity;
+                } else {
+                    return null;
```

Comment:
if not found this should not fail silently, the rest api should return 404, not found